frontend/src/components/shared/WorkflowSamplesTable/WorkflowSamplesTable.tsx

```diff
@@ -1,53 +1,61 @@
-import { Pagination, Table, TableProps } from 'antd'
+import { Button, Pagination, Table, TableProps } from 'antd'
 import { TableRowSelection } from 'antd/lib/table/interface'
 import React, { useEffect, useMemo, useState } from 'react'
-import { useAppSelector } from '../../../hooks'
+import { useAppDispatch, useAppSelector } from '../../../hooks'
 import { FMSId } from '../../../models/fms_api_models'
 import { FilterDescriptionSet, FilterKeySet, FilterSet, SetFilterFunc, SetFilterOptionFunc, SetSortByFunc, SortBy } from '../../../models/paged_items'
 import { selectLibrariesByID, selectSamplesByID } from '../../../selectors'
 import { SampleAndLibrary } from './ColumnSets'
 import { addFiltersToColumns } from './MergeColumnsAndFilters'
 import { IdentifiedTableColumnType } from './SampleTableColumns'
+import FiltersWarning from '../../filters/FiltersWarningTS'
+import getNFilters from '../../filters/getNFilters'
+
 
 export interface PaginationParameters {
 	pageNumber: number
 	pageSize: number
 	totalCount: number
-	onChangePageNumber: (pageNumber : number) => void
+	onChangePageNumber: (pageNumber: number) => void
 	onChangePageSize: (newPageSize: number) => void
 }
 
 interface WorkflowSamplesTableProps {
 	sampleIDs: FMSId[]
 	columns: IdentifiedTableColumnType<SampleAndLibrary>[]
-	filterDefinitions?: FilterDescriptionSet
-	filterKeys?: FilterKeySet
-	filters?: FilterSet
-	setFilter?: SetFilterFunc
-	setFilterOptions?: SetFilterOptionFunc
-	sortBy?: SortBy
-	setSortBy?: SetSortByFunc
-	pagination?: PaginationParameters
+	hasFilter: boolean,
+	stepNumber: number,
```

Comment:
`stepNumber` doesn't seem to be used by WorkflowSamplesTable, so this prop should be removed. Also, the samples table should remain as generic as possible, so adding a stepNumber prop isn't a good idea.